src/details/ArborX_DetailsNode.hpp

```diff
@@ -22,31 +22,35 @@ namespace ArborX
 {
 namespace Details
 {
+
+int constexpr ROPE_SENTINEL = -1;
+
 struct Node
 {
   KOKKOS_DEFAULTED_FUNCTION
   constexpr Node() = default;
 
   KOKKOS_INLINE_FUNCTION constexpr bool isLeaf() const noexcept
   {
-    return children.first == -1;
+    return left_child <= 0; // FIXME: only works with current impl
```

Comment:
So this only works because we can guarantee in the current implementation and internal node 0 is never a child. This could be changed to ` < 0`, but would need subtracting `-1` during initialization and `getPermutationIndex`.